File: mini-commerce/src/main/java/me/hero/minicommerce/item/adapter/in/web/dto/CreateItemResponse.java
package me.hero.minicommerce.item.adapter.in.web.dto;

import lombok.Value;
import me.hero.minicommerce.item.application.dto.CreateItemDto;

@Value
public class CreateItemResponse {
  private final Long id;
  private final String name;
  private final Long price;

  public CreateItemResponse(CreateItemDto dto) {
    this.id = dto.getId();
    this.name = dto.getName();
    this.price = dto.getPrice();
  }
}

File: mini-commerce/src/main/kotlin/me/hero/minicommerce/item/adapter/in/web/dto/CreateItemResponse.kt
package me.hero.minicommerce.item.adapter.`in`.web.dto

import me.hero.minicommerce.item.application.port.`in`.dto.CreateItemDto

data class CreateItemResponse(
    val id: Long,
    val name: String,
    val price: Long,
) {
    constructor(dto: CreateItemDto) : this(
        id = dto.id!!,
        name = dto.name,
        price = dto.price,
    )
}